.github/workflows/build-test.yml

```diff
@@ -730,15 +738,34 @@ jobs:
       uses: actions/upload-artifact@4cec3d8aa04e39d1a68397de0c4cd6fb9dce8ec1 # ratchet:actions/upload-artifact@v4
       with:
         name: coverage-html
-        path: target/debug/coverage/html/*
+        path: target/ff/coverage/html/*
         retention-days: 90
 
-    - name: Generate coverage report for Codecov
+    - name: Generate coverage reports for Codecov
       run: ./internal/scripts/generate-cov-report.sh --lcov
 
-    - name: Upload coverage to Codecov
+    - name: Upload Rust coverage to Codecov
       uses: codecov/codecov-action@b9fd7d16f6d7d1b5d2bec1a2887e65ceed900238 # ratchet:codecov/codecov-action@v4
       with:
-        file: target/debug/coverage/lcov.info
+        file: target/ff/coverage/lcov/iceoryx2_lcov_rust.info
+        name: iceoryx2
+        flags: Rust
         fail_ci_if_error: false
         token: ${{ secrets.CODECOV_TOKEN }}
+
+
+  # Have a sync point for all CI jobs that need to pass, will be enforced by Otterdog
+  mandatory-checks:
+    needs: [
+            static-code-analysis,
+            x86_64, x86_64_bazel,
+            x86_32,
+            coverage,
```

Comment:
What do you think about removing coverage from the list. Somethimes there were issues with the upload which were resolved later by codecove. 

The tunnel end-to-end tests might also be a candidate for a non-mandatory test. 

The lint-markdown should go in, though. 

I can do this in my open PRif you agree.